pkg/network/tracer/tracer.go

```diff
@@ -152,13 +147,20 @@ func newTracer(cfg *config.Config, telemetryComponent telemetryComponent.Compone
 			log.Warnf("NPM is explicitly enabled, so system-probe will continue with only NPM features enabled")
 		}
 	}
-
+	tt := tracerTelemetry{
+		skippedConns:         telemetryComponent.NewCounter(tracerModuleName, "skipped_conns", []string{"ip_proto"}, "Counter measuring skipped connections"),
+		expiredTCPConns:      telemetryComponent.NewCounter(tracerModuleName, "expired_tcp_conns", []string{}, "Counter measuring expired TCP connections"),
+		closedConns:          telemetry.NewStatCounterWrapper(tracerModuleName, "closed_conns", []string{"ip_proto"}, "Counter measuring closed TCP connections"),
```

Comment:
This would still use the global telemetry object instead of `telemetryComponent`. I think you can add a parameter to this to pass in `telemetryComponent`, and in `NewStatCounterWrapper` check if it is `nil` or not; if not `nil` use it to create the wrapper counter.